File: Project1-code/chapter3-work/code-Python/branch.py
import math

a = float(input())
b = float(input())
c = float(input())
delta = b*b - 4*a*c

if (a==0 and b!=0):
    print(-c/b)
    quit()

elif (a==0 and b==0 and c==0):
    print("Phuong trinh vo so nghiem")
    quit()

elif (a==0 and b==0 and c!=0):
    print("Phuong trinh vo nghiem")
    quit()
        
if (delta<0):
    print(-b/(2*a), '+', '{}i'.format(math.sqrt(abs(delta))/(2*a)))
    print(-b/(2*a), '-', '{}i'.format(math.sqrt(abs(delta))/(2*a)))
    
elif (delta>0):
        print(-b/(2*a)+math.sqrt(abs(delta))/(2*a))
        print(-b/(2*a)-math.sqrt(abs(delta))/(2*a))
    
else:
        print(-b/(2*a)+math.sqrt(abs(delta))/(2*a))

File: Project1-code/chapter3-work/code-C/branch.cpp
#include <stdio.h>
#include <math.h>
int main(){
    double a,b,c,delta;
    scanf("%lf %lf %lf",&a,&b,&c);
    delta = b*b - 4*a*c;
    if (a == 0 && b != 0){
        printf("%lf", -c/b);
        return 0;
    }else if(a == 0 && b == 0 && c == 0){
        printf("Phuong trinh vo so nghiem");
        return 0;
    }else if(a == 0 && b == 0 && c != 0){
        printf("Phuong trinh vo nghiem");
        return 0;
    }
    if(delta<0){
        printf("%lf+%lfi",-b/(2*a),sqrt(fabs(delta))/(2*a));
        printf("\n%lf-%lfi",-b/(2*a),sqrt(fabs(delta))/(2*a));
    }
    else if(delta>0){
        printf("%lf",-b/(2*a)+sqrt(fabs(delta))/(2*a));
        printf("\n%lf",-b/(2*a)-sqrt(fabs(delta))/(2*a));
    }
    else {
        printf("%lf",-b/(2*a)+sqrt(fabs(delta))/(2*a));
    }
    return 0;
}